src/pydvl/parallel/__init__.py

```diff
@@ -1,37 +1,37 @@
 """
 This module provides a common interface to parallelization backends. The list of
-supported backends is [here][pydvl.parallel.backends]. Backends can be
-selected with the `backend` argument of an instance of
-[ParallelConfig][pydvl.utils.config.ParallelConfig], as seen in the examples
-below.
+supported backends is [here][pydvl.parallel.backends]. Backends should be
+instantiated directly and passed to the respective valuation method.
 
-We use [executors][concurrent.futures.Executor] to submit tasks in parallel. The
-basic high-level pattern is
+We use executors that implement the [Executor][concurrent.futures.Executor]
+interface to submit tasks in parallel.
+The basic high-level pattern is:
 
 ```python
-from pydvl.parallel import init_executor, ParallelConfig
+from pydvl.parallel import JoblibParallelBackend
 
-config = ParallelConfig(backend="ray")
-with init_executor(max_workers=1, config=config) as executor:
+parallel_backend = JoblibParallelBackend()
+with parallel_backend.executor(max_workers=2) as executor:
     future = executor.submit(lambda x: x + 1, 1)
     result = future.result()
 assert result == 2
 ```
 
-Running a map-reduce job is also easy:
+Running a map-style job is also easy:
 
 ```python
-from pydvl.parallel import init_executor, ParallelConfig
+from pydvl.parallel import JoblibParallelBackend
 
-config = ParallelConfig(backend="joblib")
-with init_executor(config=config) as executor:
+parallel_backend = JoblibParallelBackend()
+with parallel_backend.executor(max_workers=2) as executor:
     results = list(executor.map(lambda x: x + 1, range(5)))
 assert results == [1, 2, 3, 4, 5]
 ```
 
```

Comment:
```suggestion
!!! tip "Passsing large objects"
    When running tasks which accept heavy inputs, it is important
    to first use `put()` on the object and use the returned reference
    as argument to the callable within `submit()`. For example:    
    ```python
    u_ref = parallel_backend.put(u)
    ...
    executor.submit(task, utility=u)
    ```
    Note that `task()` does not need to be changed in any way:
    the backend will `get()` the object and pass it to the function
    upon invocation.